### PR TITLE
Addressing Issue #19 for `TranscriptData` Class

### LINKED ISSUE #19: Reworking the `TranscriptData` class in `transcriptLoader.py`
Based on comments in PR #16, remaking the class used for loading the transcripts so that it aligns with the functionality and error-catching of the other classes in the pipeline would be ideal.

### PR DESCRIPTION
**Pull request by takposha**
Fixes #19 

This PR reworks the `TranscriptData` class used to process and load the caption data as raised in #19. 
It aims to make the class have a similar structure to the ones used for Topic Modelling and Question Generation, including saving the data. 
Transcript data being saved doesn't save significant computation power now, but once sentence segmentation is implemented, it will be useful to not reload and reprocess the data from scratch every time. 

Changes only significantly affect the `configData.py` and `transcriptLoader.py` scripts, with a new `.env` variable added for the option to overwrite and reprocess transcription data.

### CODE REVIEW

File: transcriptLoader.py [hunk 1]
```diff
@@ -67,96 +115,140 @@ def validateVideoFiles(self):
...
+    )
+    logging.info(f"Combined Transcript data head: {combinedTranscript.head(5)}")
+
+    return combinedTranscript
+
+
+def retrieveTranscript(config, overwrite=False):
+
+    transcriptData = TranscriptData(config)
+    if not config.overwriteTranscriptData and not overwrite:
```
Comment (pushyamig) [inline transcriptLoader.py:254]: What is the purpose of the `overwrite` variable here?  and how it is different from `overwriteTranscriptData`

Comment (takposha) [reply transcriptLoader.py:254]: It's purely for debugging purposes. In the Jupyter notebook, I can pass `overwrite` directly in the middle of running other cells without changing the config `.env` value for whatever I might be testing, It has the same functionality as `overwriteTranscriptData`. I use the `overwrite` variable in all of the retrieval functions with the same intention.

**Comment by pushyamig** [comment]
The PR looks Good and making the design pattern for transcript similar with Topic, question Generation.

File: transcriptLoader.py (hunk 1 above)
Comment (pushyamig) [reply transcriptLoader.py:254]: Can you document that? I was confused and so will next person who look at the code.

**Review by pushyamig — APPROVED** [review]
I Have one comment for documentation, after fixing it the PR is ready to go. Thanks for Keep the PR small.

File: transcriptLoader.py (hunk 1 above)
Comment (takposha) [reply transcriptLoader.py:254]: I have added this in the latest commit to all function that has this parameter, and point hem to use the corresponding `.env` variable instead.

**Pull request merged**